tensorboard/webapp/metrics/store/metrics_types.ts

```diff
@@ -40,6 +40,15 @@ import {ColumnHeader} from '../views/card_renderer/scalar_card_types';
 
 export const METRICS_FEATURE_KEY = 'metrics';
 
+/**
+ * An object which is intended to hold the min and max step within each scalar
+ * card.
+ */
+export type MinMaxStep = {
```

Comment:
These different data types already exist today but I agree this could use a cleanup. I would prefer to do this after this change has merged.
As we discussed there are ways to achieve this with `git`, however, GitHub does not have a good way of presenting PRs that are children of other open PRs. Due to this I'm going to hold off on this refactor until this series of PRs are merged.